src/main/java/org/assertj/core/internal/Diff.java

```diff
@@ -66,11 +67,29 @@
     return diff(actual.toPath(), expected, charset);
   }
 
+  @VisibleForTesting
+  public List<Delta<String>> diff(Object[] actual, Object[] expected) throws IOException {
+    return diff(
+      readerFor(arrayToStringWithDelimeter(actual, org.assertj.core.util.Compatibility.System.lineSeparator())),
+      readerFor(arrayToStringWithDelimeter(expected, org.assertj.core.util.Compatibility.System.lineSeparator())));
+  }
+
   @VisibleForTesting
   public List<Delta<String>> diff(Path actual, String expected, Charset charset) throws IOException {
     return diff(newBufferedReader(actual, charset), readerFor(expected));
   }
 
+  private String arrayToStringWithDelimeter(Object[] input, String delimeter) {
```

Comment:
Just calling .toString() on the array, puts it all on one line, which removes any benefit of using the Diff functionality, so this is used to put the array onto multiple lines.

Reply:
See if you can reuse `org.assertj.core.util.Strings.StringsToJoin` to achive the same result.